src/runtime/reactor.rs

```diff
@@ -128,8 +128,58 @@ impl Reactor {
 
     /// Block until at least one pending pollable is ready, waking a pending future.
     pub(crate) fn block_on_pollables(&self) {
+        self.check_pollables(|targets| {
+            debug_assert_ne!(
+                targets.len(),
+                0,
+                "Attempting to block on an empty list of pollables - without any pending work, no progress can be made and wasi::io::poll::poll will trap"
+            );
+            wasi::io::poll::poll(targets)
+
+        })
+    }
+
+    /// Without blocking, check for any ready pollables and wake the
+    /// associated futures.
+    pub(crate) fn nonblock_check_pollables(&self) {
+        // Lazily create a pollable which always resolves to ready.
+        use std::sync::LazyLock;
+        static READY_POLLABLE: LazyLock<Pollable> =
+            LazyLock::new(|| wasi::clocks::monotonic_clock::subscribe_duration(0));
```

Comment:
Ha, it feels like we're almost missing a built-in for this in the component model?

cc/ @lukewagner check this out

Reply:
My understanding is that there is a non-blocking mechanism to check if a waitable set contains any ready waitables in p3, so this wont be an issue there.

Reply:
Yep, that's right @pchickey: calling `waitable-set.poll` or returning the `POLL` code from a `callback` is non-blocking.